### ISSUE TITLE
CodeSniffer treats function return union type as an array declaration when type ends with "array"

### ISSUE THREAD

**Issue by alexey-m-ukolov** [squizlabs/PHP_CodeSniffer#3285]
**Describe the bug**
CodeSniffer treats function return union type as an array declaration when type ends with "array".

**Code sample**
```php
function ($item): string|array {
}
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs  --standard=Generic --sniffs=Generic.Arrays.DisallowLongArraySyntax test.php`
3. See error message displayed
```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 2 | ERROR | Short array syntax must be used to define arrays
----------------------------------------------------------------------
```

**Expected behavior**
No error is reported.

**Versions (please complete the following information):**
 - OS: Ubuntu 16.04
 - PHP: 8.0.3
 - PHPCS: 3.5.8
 - Standard: Generic

**Additional context**
Following code works as expected:
```php
function ($item): array|string {
}
```

**Comment by jrfnl** [comment]
@alexey-m-ukolov Union types were not yet supported in 3.5.8. This has been fixed in `master` and will be included in the PHPCS 3.6.0 release. I'm pretty sure you're issue is fixed now, but testing is always appreciated.

**Comment by gsherwood** [comment]
I can confirm that this is fixed in master with the new union type support, and will be released in 3.6.0.

**Comment by alexey-m-ukolov** [comment]
It worked so good with PHP 8 (apart from https://github.com/squizlabs/PHP_CodeSniffer/issues/3267) that I didn't realised it's not supported in 3.5 :)
Yes, dev-master fixes the issue for me.
Is there an ETA on 3.6?

**Comment by gsherwood** [comment]
> Is there an ETA on 3.6?

All issues are complete, so just doing final testing on attribute support before releasing. If everything goes well, it's probably a week away.
